pandas/core/arrays/datetimes.py

```diff
@@ -34,7 +34,7 @@
 _midnight = time(0, 0)
 
 
-def _to_m8(key, tz=None):
+def _to_M8(key, tz=None):
```

Comment:
isn't the conversion just (line 50)

```
In [1]: from pandas._libs.tslibs import conversion

In [2]: ts = pd.Timestamp('20181001', tz='UTC')

In [3]: np.int64(conversion.pydt_to_i8(ts)).view('M8[ns]')
Out[3]: numpy.datetime64('2018-10-01T00:00:00.000000000')

In [5]: ts.to_datetime64()
Out[5]: numpy.datetime64('2018-10-01T00:00:00.000000000')
```

Reply:
this routine should really just be in conversion.pyx

Reply:
I’m hoping we can do something clever see #24576.  Not sure we are internally consistent enough to do that yet though.